File: apollo-integration/src/test/java/com/apollographql/apollo/MockHttpCacheStore.java
package com.apollographql.apollo;

import com.apollographql.apollo.api.cache.http.HttpCacheRecord;
import com.apollographql.apollo.api.cache.http.HttpCacheRecordEditor;
import com.apollographql.apollo.api.cache.http.HttpCacheStore;

import java.io.IOException;

import org.jetbrains.annotations.NotNull;

final class MockHttpCacheStore implements HttpCacheStore {
  HttpCacheStore delegate;

  @Override public HttpCacheRecord cacheRecord(@NotNull String cacheKey) throws IOException {
    return delegate.cacheRecord(cacheKey);
  }

  @Override public HttpCacheRecordEditor cacheRecordEditor(@NotNull String cacheKey) throws IOException {
    return delegate.cacheRecordEditor(cacheKey);
  }

  @Override public void remove(@NotNull String cacheKey) throws IOException {
    delegate.remove(cacheKey);
  }

  @Override public void delete() throws IOException {
    delegate.delete();
  }
}

File: apollo-integration/src/test/kotlin/com/apollographql/apollo/MockHttpCacheStore.kt
package com.apollographql.apollo

import com.apollographql.apollo.api.cache.http.HttpCacheRecord
import com.apollographql.apollo.api.cache.http.HttpCacheRecordEditor
import com.apollographql.apollo.api.cache.http.HttpCacheStore
import java.io.IOException

internal class MockHttpCacheStore : HttpCacheStore {
  var delegate: HttpCacheStore? = null
  @Throws(IOException::class)
  override fun cacheRecord(cacheKey: String): HttpCacheRecord? {
    return delegate!!.cacheRecord(cacheKey)
  }

  @Throws(IOException::class)
  override fun cacheRecordEditor(cacheKey: String): HttpCacheRecordEditor? {
    return delegate!!.cacheRecordEditor(cacheKey)
  }

  @Throws(IOException::class)
  override fun remove(cacheKey: String) {
    delegate!!.remove(cacheKey)
  }

  @Throws(IOException::class)
  override fun delete() {
    delegate!!.delete()
  }
}